pallets/core_mods/src/migrations/mod.rs

```diff
@@ -0,0 +1,100 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod multi_key {
+    use crate::{
+        bbs_plus::BbsPlusKeys,
+        did::{keys::*, *},
+        keys_and_sigs::PublicKey,
+        util::*,
+    };
+    use alloc::collections::BTreeSet;
+    use core::{convert::identity, fmt::Debug};
+    use frame_support::{log, traits::Get, weights::Weight, *};
+    use sp_runtime::DispatchError;
+    use sp_std::{convert::TryInto, prelude::*};
+
+    pub fn migrate_unchecked_keys<T: Config + Debug>() -> Weight {
+        let mut did_keys_overridden = 0;
+        let mut dids_read = 0;
+        let mut dids_written = 0;
+        let mut did_controllers_removed = 0;
+
+        DidKeys::translate(|did, _: IncId, key: (PublicKey, u16)| {
+            did_keys_overridden += 1;
+            let key = UncheckedDidKey::new(key.0, VerRelType::try_from(key.1 & 0b1111).ok()?);
+
+            if let Ok(did_key) = key.clone().try_into() {
+                Some(did_key)
+            } else {
+                if key.ver_rels.intersects(VerRelType::CAPABILITY_INVOCATION) {
```

Comment:
Why is the motivation behind this? Comment here would help.

Reply:
Added a comment

Reply:
I don't follow, can you please add a more detailed comment?